### PR TITLE
Restoring the option to send a single embed in interaction responses.

### PR DESCRIPTION
**Pull request by Zeiin**
No need for a spare overload which is how this was split up before, and also no need to simply remove the ability to send a single without typing some array syntax around your single embed. 

Prioritizes embed arrays though, if you pass in both an embed array and a single embed, the single is discarded.

### CODE REVIEW

File: src/Discord.Net.WebSocket/Entities/Interaction/Message Components/SocketMessageComponent.cs [hunk 1]
```diff
@@ -70,15 +70,18 @@ internal override void Update(Model model)
         /// <inheritdoc/>
         public override async Task RespondAsync(
             string text = null,
-            Embed[] embeds = null,
+            Embed[] embeds = null,
```
Comment (quinchs) [inline src/Discord.Net.WebSocket/Entities/Interaction/Message Components/SocketMessageComponent.cs]: Remove trailing spaces/tabs

File: src/Discord.Net.WebSocket/Entities/Interaction/Slash Commands/SocketSlashCommand.cs [hunk 2]
```diff
@@ -52,16 +52,20 @@ internal override void Update(Model model)
         /// <inheritdoc/>
         public override async Task RespondAsync(
             string text = null,
-            Embed[] embeds = null,
+            Embed[] embeds = null,
```
Comment (quinchs) [inline src/Discord.Net.WebSocket/Entities/Interaction/Slash Commands/SocketSlashCommand.cs]: remove trailing spaces

File: src/Discord.Net.WebSocket/Entities/Interaction/Slash Commands/SocketSlashCommand.cs [hunk 3]
```diff
@@ -87,16 +91,15 @@ internal override void Update(Model model)
...
+               
             var response = new API.InteractionResponse
             {
                 Type = InteractionResponseType.ChannelMessageWithSource,
                 Data = new API.InteractionCallbackData
                 {
                     Content = text,
                     AllowedMentions = allowedMentions?.ToModel() ?? Optional<API.AllowedMentions>.Unspecified,
-                    Embeds = embeds?.Select(x => x.ToModel()).ToArray() ?? Optional<API.Embed[]>.Unspecified,
+                    Embeds = embeds?.Select(x => x.ToModel()).ToArray() ??  Optional<API.Embed[]>.Unspecified,
```
Comment (quinchs) [inline src/Discord.Net.WebSocket/Entities/Interaction/Slash Commands/SocketSlashCommand.cs]: remove trailing space

File: src/Discord.Net.WebSocket/Entities/Interaction/SocketInteraction.cs [hunk 4]
```diff
@@ -96,11 +96,12 @@ internal virtual void Update(Model model)
         ///     Responds to an Interaction with type <see cref="InteractionResponseType.ChannelMessageWithSource"/>.
         /// <para>
         ///     If you have <see cref="DiscordSocketConfig.AlwaysAcknowledgeInteractions"/> set to <see langword="true"/>, You should use
-        ///     <see cref="FollowupAsync(Discord.Embed[],string,bool,bool,Discord.AllowedMentions,Discord.RequestOptions,Discord.MessageComponent)"/> instead.
+        ///     <see cref="FollowupAsync"/> instead.
         /// </para>
         /// </summary>
         /// <param name="text">The text of the message to be sent.</param>
         /// <param name="embeds">A array of embeds to send with this response. Max 10</param>
+        /// <param name="embed">DEPRECTAED single embed to send in followup</param>
```
Comment (quinchs) [inline src/Discord.Net.WebSocket/Entities/Interaction/SocketInteraction.cs]: move the `embed` param tag to the end of the param list and remove `DEPRICATED` as its not anymore

File: src/Discord.Net.WebSocket/Entities/Interaction/SocketInteraction.cs [hunk 5]
```diff
@@ -109,13 +110,14 @@ internal virtual void Update(Model model)
...
         public abstract Task RespondAsync(string text = null, Embed[] embeds = null, bool isTTS = false,
-            bool ephemeral = false, AllowedMentions allowedMentions = null, RequestOptions options = null, MessageComponent component = null);
+            bool ephemeral = false, AllowedMentions allowedMentions = null, RequestOptions options = null, MessageComponent component = null, Embed embed = null);
 
         /// <summary>
         ///     Sends a followup message for this interaction.
         /// </summary>
         /// <param name="text">The text of the message to be sent</param>
         /// <param name="embeds">A array of embeds to send with this response. Max 10</param>
+        /// <param name="embed">DEPRECTAED single embed to send in followup</param>
```
Comment (quinchs) [inline src/Discord.Net.WebSocket/Entities/Interaction/SocketInteraction.cs]: move the `embed ` param tag to the end of the param list and remove `DEPRICATED ` as its not anymore

**Review by quinchs — CHANGES_REQUESTED** [review]

**Comment by Zeiin** [comment]
Requests addressed

**Pull request merged**